File: python/rafter_cli/utils/formatter.py
"""Output formatter with agent/human modes."""
from __future__ import annotations

_agent_mode: bool = False


def set_agent_mode(enabled: bool) -> None:
    global _agent_mode
    _agent_mode = enabled


def is_agent_mode() -> bool:
    return _agent_mode


class fmt:
    """Static formatter methods. Human mode uses Rich markup, agent mode is plain."""

    @staticmethod
    def header(text: str) -> str:
        if _agent_mode:
            return f"=== {text} ==="
        return f"\n[bold][cyan]\\u250c\\u2500[/cyan] {text} [cyan]\\u2500\\u2510[/cyan][/bold]\n"

    @staticmethod
    def success(text: str) -> str:
        if _agent_mode:
            return f"[OK] {text}"
        return f"[green]\\u2713 {text}[/green]"

    @staticmethod
    def warning(text: str) -> str:
        if _agent_mode:
            return f"[WARN] {text}"
        return f"[yellow]\\u26a0  {text}[/yellow]"

    @staticmethod
    def error(text: str) -> str:
        if _agent_mode:
            return f"[ERROR] {text}"
        return f"[red]\\u2717 {text}[/red]"

    @staticmethod
    def severity(level: str) -> str:
        upper = level.upper()
        if _agent_mode:
            return f"[{upper}]"
        styles = {
            "critical": f"[bold white on red] {upper} [/bold white on red]",
            "high": f"[bold black on yellow] {upper} [/bold black on yellow]",
            "medium": f"[white on blue] {upper} [/white on blue]",
            "low": f"[white on green] {upper} [/white on green]",
        }
        return styles.get(level, f"[{upper}]")

    @staticmethod
    def divider() -> str:
        if _agent_mode:
            return "---"
        return "[dim]" + "\u2550" * 50 + "[/dim]"

    @staticmethod
    def info(text: str) -> str:
        if _agent_mode:
            return text
        return f"[cyan]{text}[/cyan]"

File: node/src/utils/formatter.ts
import chalk from "chalk";

let agentMode = false;

export function setAgentMode(enabled: boolean): void {
  agentMode = enabled;
}

export function isAgentMode(): boolean {
  return agentMode;
}

export const fmt = {
  header(text: string): string {
    if (agentMode) return `=== ${text} ===`;
    return chalk.bold(`\n${chalk.cyan("┌─")} ${text} ${chalk.cyan("─┐")}\n`);
  },

  success(text: string): string {
    if (agentMode) return `[OK] ${text}`;
    return chalk.green(`✓ ${text}`);
  },

  warning(text: string): string {
    if (agentMode) return `[WARN] ${text}`;
    return chalk.yellow(`⚠️  ${text}`);
  },

  error(text: string): string {
    if (agentMode) return `[ERROR] ${text}`;
    return chalk.red(`✗ ${text}`);
  },

  severity(level: string): string {
    const upper = level.toUpperCase();
    if (agentMode) return `[${upper}]`;
    switch (level) {
      case "critical": return chalk.bgRed.white.bold(` ${upper} `);
      case "high": return chalk.bgYellow.black.bold(` ${upper} `);
      case "medium": return chalk.bgBlue.white(` ${upper} `);
      case "low": return chalk.bgGreen.white(` ${upper} `);
      default: return `[${upper}]`;
    }
  },

  divider(): string {
    if (agentMode) return "---";
    return chalk.gray("═".repeat(50));
  },

  info(text: string): string {
    if (agentMode) return text;
    return chalk.cyan(text);
  },
};
